src/App.js

```diff
@@ -12,6 +13,7 @@ function App() {
       </header>
       <div>
         <UserList1 userDataList={userDataList} />
+        <UserList2 userDataList={userDataList} />
```

Comment:
I couldn't quite finish an example to show you, but I thought it might be better if instead of working with the data we're given here (users), we modify that data to be more friendly to what we're trying to do. So we could transform users to something like:
```
[
  {
    client: "shaw",
    vendors: [
      {
        vendor: "shaw wireless",
        users: ["name1", "name2"]
      }
    ]
  }
]
```
That way we can make use of something call Composition in React, which leads to code looking something like:
```
<UserListHeader>
  {newDataModel.map(({client, vendors}) => (
    <ClientHeader client={client}>
      {vendors.map(({vendor, users}) => (
        <VendorHeader vendor={vendor}>
          {users.map((user) => (
            <>
              <UserInfo user={user} />
            </>
          ))}
        </VendorHeader>
      ))}
    </ClientHeader>
  ))}
</UserListHeader>
```
This usually has the benefit of logically creating components that are reusable (helpful to keep UI design consistent. 
An example of creating a component that is used in composition:
```
const ComponentName = (props) => (
  <div className={fancyCssClass}>
    <div className={otherFancyCssClass}>
      {props.customMessage}
    </div>
    {props.children}
  </div>
)
```
If you look through Endeavour, you'll see composition used a lot

Reply:
Added UserList3...  Can you take a peak